pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,458 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains utility functions for building boolean conditionals for noise models
+
+Developer note: Conditionals inherit from BooleanFn and store the condition they
+utilize in the ``condition`` attribute.
+"""
+
+from functools import partial
+from inspect import isclass, signature
+
+import pennylane as qml
+from pennylane.boolean_fn import BooleanFn
+from pennylane.wires import Wires
+
+# pylint: disable = unnecessary-lambda, too-few-public-methods
+
+
+class WiresIn(BooleanFn):
+    """A conditional for evaluating if a given wire exist in a specified set of wires.
+
+    Args:
+        wires (Union[list[int, str], Wires]): sequence of wires for building the wire set.
+
+    .. seealso:: Users are advised to use :func:`~.wires_in` for a functional construction.
+    """
+
+    def __init__(self, wires):
+        self._cond = set(wires)
+        self.condition = self._cond
```

Comment:
Why do you do this instead of `self.condition = set(wires)`?

Reply:
No specific reason, just consistency. I use `cond` as a private attribute for conditionals which I use in the computation later, whereas the `condition` attribute is a user-facing attribute that I don't access later on.